### DISCUSSION TITLE
authorization add repository helm chart

### DISCUSSION THREAD

**Discussion by rosberen** [artifacthub/hub#2761] [General]
Tell me how to register a username and password when adding a helm chart repository?

If you specify credits in the url, it outputs the error 'An error occurred when creating the repository: incorrect input: URLs with credentials are not allowed'

**Comment by tegioz** [comment]
Hi @rosberen 👋 

You can create an account from the home page, by clicking on `sign up`. Once you sign in, you can add your repositories from the control panel.

<img width="1411" alt="Screenshot 2023-02-14 at 08 00 43" src="https://user-images.githubusercontent.com/1213902/218663272-df8ae589-d169-4750-a949-ad4dcef75800.png">

<img width="437" alt="Screenshot 2023-02-14 at 08 01 23" src="https://user-images.githubusercontent.com/1213902/218663287-2102c23f-a5bb-4e32-83bd-bb16488c9fc6.png">

However, if you meant how to add a repository using the API, you'll need to create an API key from the control panel and provide it via headers in the request.

<img width="1499" alt="Screenshot 2023-02-14 at 08 06 04" src="https://user-images.githubusercontent.com/1213902/218663999-4e42aae9-116b-487b-aab1-f918fbea3bdc.png">

Hope this helps :)

**Reply by rosberen** [reply]
My repository requires authorization, where do I add a login \password to authorize the repository?
I have a private repository

**Reply by rosberen** [reply]
I found the necessary directive in the manual, solved the problem

**Reply by tegioz** [reply]
Awesome! 👍
